src/Avalonia.Controls/TopLevel.cs

```diff
@@ -878,7 +883,7 @@ private void PointerOverElementOnPropertyChanged(object? sender, AvaloniaPropert
 
         private void GlobalActualThemeVariantChanged(object? sender, EventArgs e)
         {
-            SetValue(ActualThemeVariantProperty, ((IThemeVariantHost)sender!).ActualThemeVariant, BindingPriority.Template);
+            SetCurrentValue(ActualThemeVariantProperty, ((IThemeVariantHost)sender!).ActualThemeVariant);
```

Comment:
It seems to break when:
1. User has Application.RequestedThemeVariant = Light
2. User changes Window.RequestedThemeVariant = Dark
3. User changes to Application.RequestedThemeVariant to Dark and then to Light again - Window now Light again, because SetCurrentValue has overridden value. 

`BindingPriority.Template` is a lower priority than CurrentValue, so it was used to keep global variant lower priority than variant set by user on TopLevel directly.